pkg/controller/vsphere/reconciler.go

```diff
@@ -1000,6 +1011,49 @@ func (vm *virtualMachine) getNetworkStatusList(client *vim25.Client) ([]NetworkS
 	return networkStatusList, nil
 }
 
+type attachedDisk struct {
+	device   *types.VirtualDisk
+	fileName string
+	diskMode string
+}
+
+func (vm *virtualMachine) getAttachedDisks() ([]attachedDisk, error) {
+	var attachedDiskList []attachedDisk
+	devices, err := vm.Obj.Device(vm.Context)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, disk := range devices.SelectByType((*types.VirtualDisk)(nil)) {
+		backingInfo := disk.GetVirtualDevice().Backing.(types.BaseVirtualDeviceFileBackingInfo).(*types.VirtualDiskFlatVer2BackingInfo)
+		attachedDiskList = append(attachedDiskList, attachedDisk{
+			device:   disk.(*types.VirtualDisk),
+			fileName: backingInfo.FileName,
+			diskMode: backingInfo.DiskMode,
+		})
+	}
+
+	return attachedDiskList, nil
+}
+
+func (vm *virtualMachine) detachPVDisks() error {
+	disks, err := vm.getAttachedDisks()
+	if err != nil {
+		return err
+	}
+	for _, disk := range disks {
+		// TODO (dmoiseev): should be enough, but maybe worth to check if its PV for sure
+		if disk.diskMode != "persistent" {
```

Comment:
Please use a constant from API if there is such.

Reply:
No constant for this unfortunately